design/open/DOLLAR_PULLS.md

```diff
@@ -117,3 +119,42 @@ From this view, it will be possible to create new $PULLS, re-run failed $PULLS a
 
 ![$PULLS create](diagrams/pulls-create.png)
 
+
+## Suggestion 2: Keep it minimal - Manual Triggers
+
+### How it will work
+
+Introduce a new trigger type to lakeFS actions: `manual-trigger`.
+Once defined in a lakeFS action, it will be possible to trigger the action 
+manually with a mandatory reference (branch/commit/tag).The lakeFS action 
+will run on the provided reference and the results will be available in the 
+just like any other action run.
```

Comment:
I really like manually-run actions.  They are really useful for many things, such as manually exporting data or just re-running some test that is known to be flaky.  But isn't this orthogonal to the branch protection rule?

In detail: Not sure I understand why I need to trigger an action run.  Why can't it be run as a postcommit hook or whatever?  The only difference I can see is that such an action is non-blocking for commits -- so we need to implement a nonblocking / "start-only-but-don't-wait" action type.  I think we would want it _in any case_, and the implementation for manual-trigger will in any case need to include this.  Can we separate the two and make manual-run a separate topic?

And then we don't even need to implement manual-run actions at first.

Reply:
You're right about `post-commit` hooks being enough for this task. In fact, today we don't wait for post hooks to complete, i.e. they are asynchronous. BUT - I think that some of the context is missing from this page. You can find more context [here](https://github.com/treeverse/lakeFS/pull/5152). TLDR: The requirements are that users don't want long running processes to execute with every commit. Potentially these executions are costly and should start explicitly by the user:
![image](https://user-images.githubusercontent.com/9861474/233839973-a6bde327-f456-4bbc-a2ad-834830f17361.png)